src/components/Dropdown/index.tsx

```diff
@@ -101,7 +103,9 @@ export const Dropdown = ({
         },
         onIsOpenChange({ isOpen }) {
             if (!isOpen) {
-                clearFilter();
+                if (!emptyResultMessage) {
+                    clearFilter();
+                }
             }
```

Comment:
Would it work writing it this way since the clearFilter is the only thing we're calling for !isOpen and !emptyResultMessage
```suggestion
if (!isOpen && !emptyResultMessage) clearFilter();
```